src/test/kotlin/pt/up/fe/ni/website/backend/controller/AuthControllerTest.kt

```diff
@@ -180,6 +201,367 @@ class AuthControllerTest @Autowired constructor(
         }
     }
 
+    @NestedTest
+    @DisplayName("POST /auth/password/recovery")
+    inner class RecoverPasswordRequest {
+        var documentation: ModelDocumentation = PayloadRecoverPassword()
+
+        @BeforeEach
+        fun setup() {
+            repository.save(testAccount)
+        }
+
+        @Test
+        fun `should return empty if email is not found`() {
+            mockMvc.perform(
+                post("/auth/password/recovery")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(objectMapper.writeValueAsString(mapOf("email" to "dont@exist.com")))
+            )
+                .andExpectAll(
+                    status().isOk(),
+                    jsonPath("$.recovery_url").doesNotExist()
+                )
+                .andDocument(
+                    documentation,
+                    "Recover password",
+                    "This endpoint operation allows the recovery of the password of an account, " +
+                        "sending an email with a link to reset the password.",
+                    documentRequestPayload = true
+                )
+        }
+
+        @Test
+        fun `should return password recovery link`() {
+            mockMvc.perform(
+                post("/auth/password/recovery")
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .content(objectMapper.writeValueAsString(mapOf("email" to testAccount.email)))
+            )
+                .andExpectAll(
+                    status().isOk(),
+                    jsonPath("$.recovery_url").exists()
+                ).andDocument(
+                    documentation,
+                    "Recover password",
+                    "This endpoint operation allows the recovery of the password of an account, " +
+                        "sending an email with a link to reset the password.",
+                    documentRequestPayload = true
+                )
+        }
```

Comment:
As I explained before, this is not the expected behavior. Both cases will return empty and ok status.

What you should do is test the request returns empty and ok both times, but also to test if the first situation does not call the generate token function. I think the easiest way would be by mocking JwtClaimsSet and verifying if some of those functions do not get called.

Reply:
Same as above, this could be as is for now and changed when the email service is implemented.
Can leave a comment to remind that.